core/src/main/java/org/apache/calcite/rel/rules/ReduceExpressionsRule.java

```diff
@@ -346,15 +345,6 @@ public JoinReduceExpressionsRule(Class<? extends Join> joinClass,
           matchNullability)) {
         return;
       }
-      if (RelOptUtil.forceEquiJoin(join)) {
-        final JoinInfo joinInfo =
-            JoinInfo.of(join.getLeft(), join.getRight(), expList.get(0));
-        if (!joinInfo.isEqui()) {
-          // This kind of join must be an equi-join, and the condition is
-          // no longer an equi-join. SemiJoin is an example of this.
-          return;
-        }
-      }
```

Comment:
We can not remove this, the behavior would be non-consistent, how about other systems use this rule and their join does not support non-equi conditions ?

Reply:
Then the other systems should update their own implementation, as EquiJoin is deprecated.